rust/foxglove/src/channel_builder.rs

```diff
@@ -83,9 +84,6 @@ impl ChannelBuilder {
     /// Builds a [`Channel`].
     ///
     /// `T` must implement [`Encode`].
-    ///
-    /// Returns [`FoxgloveError::DuplicateChannel`] if a channel with the same topic already
-    /// exists.
     pub fn build<T: Encode>(mut self) -> Result<Channel<T>, FoxgloveError> {
```

Comment:
This can't fail now, should we not return a result type here?

Reply:
Done. Note this resulted in changes to the python classes too (which is good; our examples weren't catching anyway). I added a unit test that should catch the expectation there.